### PR TITLE
Perpetual swaps

### PR DESCRIPTION
**Pull request by tiltom**

### CODE REVIEW

File: src/app/pages/PerpetualPage/components/RecentTradesTable/types.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export enum TradePriceChange {
+  'UP' = 'up',
+  'DOWN' = 'down',
+  'NO_CHANGE' = 'noChange',
+}
+
+export enum TradeType {
+  'SELL' = 'sell',
+  'BUY' = 'buy',
+}
```
Comment (creed-victor) [inline src/app/pages/PerpetualPage/components/RecentTradesTable/types.ts:10]:
```suggestion
export enum TradePriceChange {
  UP = 'up',
  DOWN = 'down',
  NO_CHANGE = 'noChange',
}

export enum TradeType {
  SELL = 'sell',
  BUY = 'buy',
}
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

File: src/app/components/HeaderLabs/index.module.scss [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+.logo {
+  width: auto;
+  height: 35px;
+  margin: 0 0 0 1rem;
+
+  screem xl {
+    width: 216px;
+    height: 53px;
+    margin: 0;
+  }
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/components/HeaderLabs/index.module.scss:38]: good catch

**Review by soulBit — CHANGES_REQUESTED** [review]

File: src/app/hooks/useSendContractTx.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ import { ContractName } from 'utils/types/contracts';
 import { useAccount } from './useAccount';
 import { Nullable } from 'types';
 import { gasLimit } from '../../utils/classifiers';
+import { selectWalletProvider } from '../containers/WalletProvider/selectors';
 
 export interface TransactionOptions {
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/hooks/useSendContractTx.ts:22]: Yeah we should make a definitive decision on that, we have 213 instances of interfaces that don't start with I :sweat_smile:
We should probably handle it all in one swoop.

[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/hooks/useSendContractTx.ts:22]: Will not change it here for now.

File: src/app/pages/MarginTradePage/components/AdvancedSettingDialog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { FormGroup } from 'app/components/Form/FormGroup';
+import { Slider } from 'app/components/Form/Slider';
+
+// import { translations } from '../../../BuySovPage/components/Slider';
+// import { useMaintenance } from 'app/hooks/useMaintenance';
+// import { TradingPosition } from 'types/trading-position';
+// import { discordInvite } from 'utils/classifiers';
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/pages/MarginTradePage/components/AdvancedSettingDialog/index.tsx:12]: Ah shit. We totally forgot about this stuff. I'll need to revert all the margin page changes to the development state.

[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/pages/MarginTradePage/components/AdvancedSettingDialog/index.tsx]: Fixed it reverted the whole MarginTradePage to development.

**Review by creed-victor — DISMISSED** [review]

**Review by creed-victor — APPROVED** [review]

**Review by creed-victor — DISMISSED** [review]

**Review by creed-victor — APPROVED** [review]

**Comment by tiltom** [comment]
Merged in https://github.com/DistributedCollective/Sovryn-frontend/pull/2136

**Pull request closed without merging**